### ISSUE TITLE
Decouple `build` folder name from metal

### ISSUE THREAD

**Issue by vtangTT** [tenstorrent/tt-metal#8767]
We want to let devs decide what the build folder is called.

There are hardcoded paths in metal right now assuming the build folder is called `build`

Inside`tt_metal/jit_build/build.cpp` we have stuff like:

```this->lflags_ += "-T" + env_.root_ + "build/hw/toolchain/idle-erisc.ld ";```

Inside `tt_metal/tools/profiler/common.py` we have:

```PROFILER_BIN_DIR = TT_METAL_HOME / "build/tools/profiler/bin```

CC: @TT-billteng @pgkeller @mo-tenstorrent

**Comment by vtangTT** [comment]
Yes, I think the idea is to pass in a build folder as an env var when using Tracy with named build folders. Correct me if I'm wrong @TT-billteng.

Would this be a very invasive change?

**Comment by TT-billteng** [comment]
> Yes, I think the idea is to pass in a build folder as an env var when using Tracy with named build folders. Correct me if I'm wrong @TT-billteng.
> 
> Would this be a very invasive change?

I would hope we could pass it in as an option somehow

**Comment by tt-rkim** [comment]
Is this done?

**Comment by vtangTT** [comment]
I think we still need to decouple the profiler files. @mo-tenstorrent do you have cycles for this?

**Comment by mo-tenstorrent** [comment]
This is a minor change, providing an option for `tracy.py` which is the only user of this location is no problem. I am just thinking if there is any way we can extract info on the latest build folder so `tracy.py` can use that by default, instead of just `build` as default. `tracy.py` is getting pretty cli option heavy it would be nice to be able to have smarts here.

**Comment by TT-billteng** [comment]
is this done?
